src/main/java/chess/board/ChessBoard.java

```diff
@@ -0,0 +1,122 @@
+package chess.board;
+
+import java.util.Arrays;
+import java.util.Comparator;
+import java.util.List;
+import java.util.stream.Collectors;
+import java.util.stream.IntStream;
+
+public class ChessBoard {
+    private static final int MIN_ROW_NUMBER = 1;
+    private static final int MAX_ROW_NUMBER = 8;
+    private static final int ROW_INDEX = 1;
+    private static final int COLUMN_INDEX = 0;
+    private static final boolean FIRST_TRY = false;
+    
+    private final List<RowPieces> chessBoard;
+    
+    public ChessBoard(List<RowPieces> chessBoard) {
+        this.chessBoard = chessBoard;
+    }
+    
+    public static ChessBoard create(){
+        return new ChessBoard(initChessBoard());
+    }
+    
+    private static List<RowPieces> initChessBoard() {
+        return IntStream.rangeClosed(MIN_ROW_NUMBER, MAX_ROW_NUMBER)
+                .mapToObj(RowPieces::new)
+                .sorted(Comparator.reverseOrder())
+                .collect(Collectors.toList());
+    }
+    
+    public void move(String sourceCoordinate, String destinationCoordinate) {
+        List<Integer> parsedSourceCoordinate = parseCoordinate(sourceCoordinate);
+        List<Integer> parsedDestinationCoordinate = parseCoordinate(destinationCoordinate);
+        RowPieces rowPiecesContainsSourcePiece = findRowPiecesByRow(parsedSourceCoordinate.get(ROW_INDEX));
+        RowPieces rowPiecesContainsDestinationPiece = findRowPiecesByRow(parsedDestinationCoordinate.get(ROW_INDEX));
```

Comment:
위가 Coodinate 라는 타입이라면 아래와 같이 변경해볼 수 있겠네요

```suggestion
        RowPieces rowPiecesContainsSourcePiece = findRowPiecesByRow(parsedSourceCoordinate);
        RowPieces rowPiecesContainsDestinationPiece = findRowPiecesByRow(parsedDestinationCoordinate);
        
    private RowPieces findRowPiecesByCoodinate(Coordinate coordinate) {
        return chessBoard.stream()
                .filter(rowPieces -> rowPieces.isSameCoodinate(coordinate))
                .findFirst()
                .orElseThrow(() -> new IllegalArgumentException("해당 좌표는 존재하지 않습니다."));
    }
```

Reply:
네! 위에서도 말씀 드렸지만, ChessBoard에서 Coordinate로 기능할 수 있는 부분들 전부 리팩토링 해보겠습니다.